motion_computation/CMakeLists.txt

```diff
@@ -0,0 +1,122 @@
+cmake_minimum_required(VERSION 2.8.3)
+project(motion_computation)
+
+## Compile as C++11, supported in ROS Kinetic and newer
+add_compile_options(-std=c++11)
+
+## Find catkin macros and libraries
+## if COMPONENTS list like find_package(catkin REQUIRED COMPONENTS xyz)
+## is used, also find other catkin packages
+find_package(catkin REQUIRED COMPONENTS
+  autoware_msgs
+  cav_msgs
+  roscpp
+  carma_utils
+  motion_predict
+)
+
+###################################
+## catkin specific configuration ##
+###################################
+## The catkin_package macro generates cmake config files for your package
+## Declare things to be passed to dependent projects
+## INCLUDE_DIRS: uncomment this if your package contains header files
+## LIBRARIES: libraries you create in this project that dependent projects also need
+## CATKIN_DEPENDS: catkin_packages dependent projects also need
+## DEPENDS: system dependencies of this project that dependent projects also need
+catkin_package(
+   INCLUDE_DIRS include
+#  LIBRARIES motion_computation
+ CATKIN_DEPENDS autoware_msgs cav_msgs roscpp carma_utils motion_predict
+#  DEPENDS system_lib
+)
+
+###########
+## Build ##
+###########
+
+## Specify additional locations of header files
+## Your package locations should be listed before other locations
+include_directories(
+  include
+  ${catkin_INCLUDE_DIRS}
+)
+
+add_library(motion_computation_worker_lib
+ src/motion_computation_worker.cpp
+ src/main.cpp
+ src/motion_computation_node.cpp
+ )
+
+add_dependencies(motion_computation_worker_lib ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
+
+add_executable(${PROJECT_NAME}_node
+ src/main.cpp 
+ src/motion_computation_node.cpp
+ src/motion_computation_worker.cpp
+)
```

Comment:
You have multiple instances of your cpp files listed here and in the above library. You should only ever list a .cpp file once in your cmake. In this case you want the worker.cpp in the library and the node/main.cpp files in the executable.